lib/tls.js

```diff
@@ -75,7 +75,7 @@ exports.checkServerIdentity = function checkServerIdentity(host, cert) {
   // Create regexp to much hostnames
   function regexpify(host, wildcards) {
     // Add trailing dot (make hostnames uniform)
-    if (!/\.$/.test(host)) host += '.';
+    if (!host || !host.endsWith('.')) host += '.';
```

Comment:
Can it really? In that case, the old code would have produced `undefined.`. Does that code path really exist?